guava/src/test/java/com/fasterxml/jackson/datatype/guava/MultiMapOrderMpEntriesByKeys7Test.java

```diff
@@ -0,0 +1,157 @@
+package com.fasterxml.jackson.datatype.guava;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.collect.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MultiMapOrderMpEntriesByKeys7Test extends ModuleTestBase {
+/*
+    /**********************************************************
+    /* Set up
+    /**********************************************************
+     */
+
+    static class UncomparableBean {
+        public String value;
+
+        public UncomparableBean(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+    }
+
+    static class PolymorphicWrapperBean {
+        @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type", visible = true)
+        public Object aProperty;
+
+        @Override
+        public String toString() {
+            return aProperty.toString();
+        }
+    }
+
+    /*
+    /**********************************************************
+    /* Tests
+    /**********************************************************
+     */
+
+    private final ObjectMapper MAPPER = mapperWithModule();
+
+    public void testMultimapSerializeOrderedByKey() throws Exception {
+        final Multimap<Object, Object> multimap = HashMultimap.create();
+        multimap.put("c_key", 1);
+        multimap.put("d_key", 1);
+        multimap.put("a_key", 1);
+        multimap.put("e_key", 1);
+        multimap.put("b_key", 1);
+
+        String jsonStr = MAPPER.writer()
+            .with(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
+            .writeValueAsString(multimap);
+
+        assertEquals(a2q("{'a_key':[1],'b_key':[1],'c_key':[1],'d_key':[1],'e_key':[1]}"), jsonStr);
+    }
+
+    public void testMultimapSerializeUnorderedByKey() throws Exception {
+        final Multimap<Object, Object> multimap = HashMultimap.create();
+        multimap.put("c_key", 1);
+        multimap.put("d_key", 1);
+        multimap.put("a_key", 1);
+        multimap.put("e_key", 1);
+        multimap.put("b_key", 1);
+
+        String jsonStr = MAPPER.writer()
+            .without(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
+            .writeValueAsString(multimap);
+
+        assertNotSame(a2q("{'a_key':[1],'b_key':[1],'c_key':[1],'d_key':[1],'e_key':[1]}"), jsonStr);
+    }
+
+
+    public void testMultimapSerializeWithNullKeyFailure() throws Exception {
+        final Multimap<Object, Object> multimap = HashMultimap.create();
+        multimap.put("c_key", 1);
+        multimap.put("d_key", 1);
+        multimap.put(null, 1);
```

Comment:
Oh. So, `null`s do work even with default impl? Hmmmh.

Reply:
Ok, although I have often disagreed with Guava's insistence on null-avoidance, it makes sense here.

But... if `HashMultimap` allows it, we do still have an issue, kind of. So I now better understand why you had NPE handling there (should have assumed there was a solid reason for it).

I am really not quite sure what to do with it, to be honest: serialization does actually deal with `null` keys otherwise.

Reply:
Actually, let me do this: I'll merge things as they are, and then have a look at NPE.

Reply:
Will do. Thank you -- once again! -- for the contribution!